test/Performance/BuildPerformanceTest.cs

```diff
@@ -0,0 +1,249 @@
+
+        public void IncrementalSkipSingleProject(TestInstance instance)
+        {
+            Build(instance.TestRoot);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    Build(instance.TestRoot);
+                }
+            }
+        }
+
+        [Benchmark]
+        public void BuildAllInGraph_SingleTargetGraph() => BuildAllInGraph(CreateTestInstances(SingleTargetGraph));
+        [Benchmark]
+        public void BuildAllInGraph_TwoTargetGraph() => BuildAllInGraph(CreateTestInstances(TwoTargetGraph));
+        [Benchmark]
+        public void BuildAllInGraph_TwoTargetGraphLarge() => BuildAllInGraph(CreateTestInstances(TwoTargetGraphLarge));
+
+        public void BuildAllInGraph(TestInstance[] instances)
+        {
+            var instance = instances[0];
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    Build(instance.TestRoot);
+                }
+                foreach (var i in instances)
+                {
+                    RemoveBin(i.TestRoot);
+                }
+            }
+        }
+
+        [Benchmark]
+        public void IncrementalSkipAllInGraph_SingleTargetGraph() =>
+            IncrementalSkipAllInGraph(CreateTestInstances(SingleTargetGraph));
+
+        [Benchmark]
+        public void IncrementalSkipAllInGraph_TwoTargetGraph() =>
+            IncrementalSkipAllInGraph(CreateTestInstances(TwoTargetGraph));
+
+        [Benchmark]
+        public void IncrementalSkipAllInGraphh_TwoTargetGraphLarge() =>
+            IncrementalSkipAllInGraph(CreateTestInstances(TwoTargetGraphLarge));
+
+        public void IncrementalSkipAllInGraph(TestInstance[] instances)
+        {
+            var instance = instances[0];
+
+            Build(instance.TestRoot);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    Build(instance.TestRoot);
+                }
+            }
+        }
+
+        [Benchmark]
+        public void IncrementalRebuildWithRootChangedInGraph_SingleTargetGraph() =>
+            IncrementalRebuildWithRootChangedInGraph(CreateTestInstances(SingleTargetGraph));
+
+        [Benchmark]
+        public void IncrementalRebuildWithRootChangedInGraph_TwoTargetGraph() =>
+            IncrementalRebuildWithRootChangedInGraph(CreateTestInstances(TwoTargetGraph));
+
+        [Benchmark]
+        public void IncrementalRebuildWithRootChangedInGraph_TwoTargetGraphLarge() =>
+            IncrementalRebuildWithRootChangedInGraph(CreateTestInstances(TwoTargetGraphLarge));
+
+        public void IncrementalRebuildWithRootChangedInGraph(TestInstance[] instances)
+        {
```

Comment:
This test doesn't actually change the root, but instead just deletes the bin directory.  Is this the scenario you're trying to test, or do you want to actually test changing some code in the root project?  I would think that the performance characteristics might be different because the bin directory will exist in one case but not the other.